### PR TITLE
ultrastardx: 2017.8.0 -> 2019-01-07

### PR DESCRIPTION
**Pull request by teto**
follow up of the fix provided in https://github.com/UltraStar-Deluxe/USDX/issues/462
to use pkg-config to find lua.

cc @vcunat https://github.com/NixOS/nixpkgs/commit/fbda7ca802b

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/games/ultrastardx/default.nix [hunk 1]
```diff
@@ -7,26 +7,24 @@ let
...
     pcre portaudio freetype
     SDL2 SDL2_image SDL2_gfx SDL2_mixer SDL2_net SDL2_ttf
-    sqlite lua zlib libX11 libGLU libGL ffmpeg
+    lua sqlite zlib libX11 libGLU libGL ffmpeg
   ];
 
 in stdenv.mkDerivation rec {
   pname = "ultrastardx";
-  version = "2017.8.0";
+  version = "07012019";
```
Comment (flokli) [inline pkgs/games/ultrastardx/default.nix]: You're sure about that version string?

Comment (teto) [reply pkgs/games/ultrastardx/default.nix]: fixed as per the manual ``` If a package is not a release but a commit from a repository, then the version part of the name must be the date of that (fetched) commit. The date must be in "YYYY-MM-DD" format. Also append "unstable" to the name - e.g., "pkgname-unstable-2014-09-23". ```

Comment (flokli) [reply pkgs/games/ultrastardx/default.nix]: I saw `version = "unstable-YYYY-MM-DD";` more often than `pname = "foo-unstable";`. Can you change this?

Also, the commit message still contains the old version number.

Comment (teto) [reply pkgs/games/ultrastardx/default.nix]: should be fine now

**Review by flokli — APPROVED** [review]

**Comment by mweinelt** [comment]
Ran nix-review on this pr and tested the resulting binary.

LGTM.

**Pull request merged**